File: aspnet/web-forms/overview/older-versions-getting-started/master-pages/control-id-naming-in-content-pages-vb/samples/sample12.vb
Imports System.Runtime.CompilerServices

Public Module PageExtensionMethods
 <Extension()> _
  Public Function FindControlRecursive(ByVal ctrl As Control, ByVal controlID As String) As Control
 If String.Compare(ctrl.ID, controlID, True) = 0 Then
 ' We found the control!
 Return ctrl
 Else
 ' Recurse through ctrl's Controls collections
 For Each child As Control In ctrl.Controls
 Dim lookFor As Control = FindControlRecursive(child, controlID)

 If lookFor IsNot Nothing Then
 Return lookFor  ' We found the control
 End If
 Next

 ' If we reach here, control was not found
 Return Nothing
 End If
 End Function
End Module

File: aspnet/web-forms/overview/older-versions-getting-started/master-pages/control-id-naming-in-content-pages-cs/samples/sample12.cs
using System;
using System.Web;
using System.Web.UI;

public static class PageExtensionMethods
{
	public static Control FindControlRecursive(this Control ctrl, string controlID)
	{
		if (string.Compare(ctrl.ID, controlID, true) == 0)
		{
			// We found the control!
			return ctrl;
		}
		else
		{
			// Recurse through ctrl's Controls collections
			foreach (Control child in ctrl.Controls)
			{
				Control lookFor = FindControlRecursive(child, controlID);

				if (lookFor != null)
					return lookFor;  // We found the control
			}

			// If we reach here, control was not found
			return null;
		}
	}
}